src/impl/realm_coordinator.cpp

```diff
@@ -300,6 +300,21 @@ void RealmCoordinator::get_realm(Realm::Config config,
     callback(realm, nullptr);
 }
 
+#if REALM_ENABLE_SYNC
+std::shared_ptr<AsyncOpenTask> RealmCoordinator::get_synchronized_realm(Realm::Config config)
+{
+    if (config.sync_config) {
+        std::unique_lock<std::mutex> lock(m_realm_mutex);
+        set_config(config);
+        create_sync_session(!File::exists(m_config.path));
```

Comment:
I don't think this check is correct? If the file already exists but no sync_session exists then the returned AsyncOpenTask will just do nothing.

Reply:
Not sure I understand? That method will always create the sync session, the boolean just forces a client reset (and thus faster download)? It is the same implementation as in your implementation of `get_shared_realm(config, callback)`

Reply:
Yeah that comment's just wrong.